### ISSUE TITLE
Train with multiply GPU

### ISSUE THREAD

**Issue by tfiliano** [graphdeco-inria/gaussian-splatting#694]
Is there any way to do it?

someone could point the way I need to follow to do it?

Thanks

**Comment by jaco001** [comment]
As I know - they must rewrite cuda code part for it. Training part for now isn't bad, Only reason for rewrite code is very big datasets, but  gaussians can be trained only from few pics.... so duno if authors do that. ;)

**Comment by LoickCh** [comment]
#241

**Comment by tfiliano** [comment]
> #241

Yeah I have spotted that one, I can try to implement it, but Python, Cuda, etc is not my world

**Comment by tfiliano** [comment]
> could Dask be possibly integrated to distribute the job?

I was looking at this https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html
Once the project already uses Pytorch, I was thinking that it could be one possibility.

But yeah.. I will need some time reading the source code and trying to understand it to see how and where to apply this nn.parallel

**Comment by TarzanZhao** [comment]
This project enables distributed gaussian splatting training: https://github.com/nyu-systems/Grendel-GS

**Comment by Moreor1ess** [comment]
Do you know to choose GPU? The default is to run on GPU 0, but I want it to run on GPU 1
